apps/main/src/store/createPoolsSlice.ts

```diff
@@ -348,32 +347,21 @@ const createPoolsSlice = (set: SetState<State>, get: GetState<State>): PoolsSlic
 
       sliceState.setStateByActiveKey('currencyReserves', getChainPoolIdActiveKey(chainId, pool.id), result)
     },
-    fetchPoolsChunkRewardsApy: async (chainId, poolDatas) => {
-      const { results } = await PromisePool.for(poolDatas).process(
-        async (poolData: PoolData) => await networks[chainId].api.pool.poolAllRewardsApy(chainId, poolData.pool)
-      )
-      // create mapper
-      const rewardsApyMapper: RewardsApyMapper = cloneDeep(get()[sliceKey].rewardsApyMapper[chainId] ?? {})
-      for (const idx in results) {
-        const rewardsApy = results[idx]
-        rewardsApyMapper[rewardsApy.poolId] = rewardsApy
-      }
-      get()[sliceKey].setStateByActiveKey('rewardsApyMapper', chainId.toString(), rewardsApyMapper)
-      return rewardsApyMapper
-    },
     fetchPoolsRewardsApy: async (chainId, poolDatas) => {
       log('fetchPoolsRewardsApy', chainId, poolDatas.length)
+      let rewardsApyMapper: RewardsApyMapper = { ...get()[sliceKey].rewardsApyMapper[chainId] }
```

Comment:
I merged the 2 functions so we don't need to reclone this variable every iteration